design/one-pager-resource-connectivity-mvp.md

```diff
@@ -0,0 +1,656 @@
+    clusterIpv4CidrBlock: 10.0.0.0/8
+    nodeIpv4CidrBlock: 172.16.1.0/24
+    servicesIpv4CidrBlock: 192.168.0.0/24
+    tpuIpv4CidrBlock: 172.16.2.0/24
+  masterAuthorizedNetworksConfig:
+    enabled: true
+    cidrBlocks:
+      displayName: coolcidr
+      cidrBlock: 93.80.0.0/16
+  maintenancePolicy: object (MaintenancePolicy)
+    window:
+      dailyMaintenanceWindow:
+        startTime: 01:30
+  defaultMaxPodsConstraint:
+    maxPodsPerNode: 254
+  resourceUsageExportConfig:
+    bigqueryDestination:
+      datasetId: mycooldataset
+    enableNetworkEgressMetering: true
+    consumptionMeteringConfig:
+      enabled: true
+  privateClusterConfig:
+    enablePrivateNodes: true
+    enablePrivateEndpoint: true
+    masterIpv4CidrBlock: 172.16.0.0/16
+  initialClusterVersion: 1.13
+  enableTpu: false
+```
+
+## Proposal
+
+Minimum viable support for resource connectivity can be enabled by ensuring
+three things.
+
+* All external resources required to configure connectivity have corresponding
+  Crossplane managed resources, such that the cloud administrator does not need
+  to leave Crossplane to configure them. For example, if a `KubernetesCluster`
+  can only connect to a `MySQLInstance` in the same VPC network then Crossplane
+  must be able to represent said VPC network as a managed resource.
+* All external resource settings required to configure connectivity must be
+  exposed in the configuration of their associated managed resource.
+* All managed resource settings required to configure connectivity must be
+  exposed in the configuration of their associated resource class.
+
+With the above in place resource connectivity between a `KubernetesCluster` and
+a `MySQLInstance` can be configured roughly as follows:
+
+1. The cloud administrator creates any managed resources necessary to connect
+   a Kubernetes cluster to a MySQL database in their cloud of choice, for
+   example they create a VPC network for both external resources to live in.
+1. The cloud administrator creates resource classes to be used when creating
+   `KubernetesCluster` and `MySQLInstance` resource claims against their cloud
+   of choice. These resource classes specify the necessary configuration to
+   ensure any managed resources backing the aforementioned claims can connect,
+   for example ensuring they're both configured to use the aforementioned VPC
+   network.
+1. The app operator creates their `KubernetesCluster` and `MySQLInstance`
+   resource claims, which either explicitly reference or default to the resource
+   classes created in step 2. Because their underlying managed resources are
+   appropriately configured, they can now communicate!
+
+The remainder of this proposal works through this scenario in each cloud
+provider, highlighting the changes necessary to enable connectivity.
+
+### Resource References
+
+It's typical for Kubernetes resources to refer to each other. The [Kubernetes
+API conventions] states:
```

Comment:
Nitpick: `conventions` and `states` don't match, grammatically. Would suggest referring to the conventions document using a singular form (`Kubernetes API conventions document`) or using the singular form for `state` (`The [Kubernetes API conventions] state:`).